conftest.py

```diff
@@ -604,40 +586,53 @@ def old_workflow_project(request, tmp_path_factory):
 
 
 @pytest.fixture
-def old_dataset_project(tmp_path_factory):
+def old_dataset_project(tmp_path):
     """Prepares a testing repo created by old version of renku."""
+    from renku import LocalClient
+
     name = "old-datasets-v0.9.1.git"
-    base_path = tmp_path_factory.mktemp(name)
+    base_path = tmp_path / name
     repository = clone_compressed_repository(base_path=base_path, name=name)
 
     os.chdir(repository.working_dir)
 
-    yield repository
-
-    shutil.rmtree(base_path)
+    yield LocalClient(path=repository.working_dir)
 
 
 @pytest.fixture
-def old_repository_with_submodules(request, tmpdir_factory):
+def old_repository_with_submodules(request, tmp_path):
     """Prepares a testing repo that has datasets using git submodules."""
     name = "old-datasets-v0.6.0-with-submodules"
     base_path = Path(__file__).parent / "tests" / "fixtures" / f"{name}.tar.gz"
 
-    working_dir = tmpdir_factory.mktemp(name)
+    working_dir = tmp_path / name
 
     with tarfile.open(str(base_path), "r") as repo:
-        repo.extractall(working_dir.strpath)
+        repo.extractall(working_dir)
 
     repo_path = working_dir / name
     repo = Repo(repo_path)
 
-    os.chdir(repo_path.strpath)
+    os.chdir(repo_path)
     yield repo
 
-    shutil.rmtree(repo_path.strpath)
+    shutil.rmtree(repo_path)
```

Comment:
I think we should chdir back to the original directory to make sure we end up in a good state.

Reply:
I've changed them all to use the `chdir` context manager. However, I believe working directory gets reset at the beginning of the each test and this should not matter much for function-scoped fixtures.